File: babydata/src/main/java/com/zero/babydata/entity/SleepRecord.java
package com.zero.babydata.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "Sleep_Records",
        foreignKeys = @ForeignKey(entity = BabyInfo.class,
                parentColumns = "babyId",
                childColumns = "babyId",
                onDelete = ForeignKey.CASCADE), indices = {@Index(value = "babyId")})
public class SleepRecord {
    @PrimaryKey(autoGenerate = true)
    public int sleepId;
    public int babyId;
    public Long sleepStart;
    public Long sleepEnd;
    public int sleepDuration;
    public String note;
    public Long createdAt;
}

File: babydata/src/main/java/com/zero/babydata/entity/SleepRecord.kt
package com.zero.babydata.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "Sleep_Records",
    foreignKeys = [ForeignKey(
        entity = BabyInfo::class,
        parentColumns = ["babyId"],
        childColumns = ["babyId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index(value = ["babyId"])]
)
data class SleepRecord(
    @PrimaryKey(autoGenerate = true)
    var sleepId: Int = 0,
    var babyId: Int = 0,
    var sleepStart: Long = 0L,
    var sleepEnd: Long = 0L,
    var sleepDuration: Long = 0L,
    var note: String = "",
    var createdAt: Long = 0L
)
